core/merginapi.cpp

```diff
@@ -3175,12 +3224,20 @@ bool MerginApi::writeData( const QByteArray &data, const QString &path )
   createPathIfNotExists( path );
   if ( !file.open( QIODevice::WriteOnly ) )
   {
+    CoreUtils::log( "writeData ", "Failed to open file for writing: " + path );
     return false;
   }
 
-  file.write( data );
+  qint64 bytesWritten = file.write( data );
   file.close();
 
+  // not enough space on file system
+  if ( data.size() != bytesWritten )
+  {
+    CoreUtils::log( "writeData ", "Failed to write content into: " + path );
+    return false;
```

Comment:
https://github.com/MerginMaps/input/blob/780622dacf156cad443caa5fa6fd0089c21e2ae9/core/merginapi.cpp#L3178 does not check the return value of this function, so the sync is not failed.